### PR TITLE
DI support for storage session

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

**Comment by Ivan-L** [comment]
Thank you for this @danielmarbach 👌

File: src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/StorageSessionFactory.cs [hunk 1]
```diff
@@ -5,13 +5,20 @@
...
         {
+            this.currentSharedTransactionalBatchHolder = currentSharedTransactionalBatchHolder;
             this.containerHolder = containerHolder;
         }
 
-        public Task<CompletableSynchronizedStorageSession> OpenSession(ContextBag contextBag) => Task.FromResult<CompletableSynchronizedStorageSession>(new StorageSession(containerHolder, contextBag, true));
+        public Task<CompletableSynchronizedStorageSession> OpenSession(ContextBag contextBag)
+        {
+            var storageSession = new StorageSession(containerHolder, contextBag, true);
+            currentSharedTransactionalBatchHolder?.SetCurrent(storageSession);
```
Comment (boblangley) [inline src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/StorageSessionFactory.cs:17]: Why would this ever be null?

If I am reading `SynchronizedStorage` right, we always create this and pass it into the factory.

https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/35/files#diff-90cae606f9a2e42dd938a294ac38d378R18

Comment (danielmarbach) [reply src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/StorageSessionFactory.cs:17]: It is null in the persistence component tests. We don't need this there

**Comment by danielmarbach** [comment]
Can't merge because there is no approval @SeanFeldman @boblangley

**Review by SeanFeldman — APPROVED** [review]

**Comment by SeanFeldman** [comment]
Approved but would need to go over this again. My brain melted. Several times.

**Comment by danielmarbach** [comment]
Happy to do so Sean.

**Pull request merged**